pdf/pdf.go

```diff
@@ -48,21 +48,35 @@ func NormalizePageRange(pageCount int, pageRange string) (*string, *int, error)
 			} else if strings.HasPrefix(pageRangeParts[pageRangePartI], "r") {
 				parsedPageNumber, err := strconv.Atoi(strings.TrimPrefix(pageRangeParts[pageRangePartI], "r"))
 				if err != nil {
+					if ignoreInvalidPages {
+						continue
+					}
 					return nil, nil, fmt.Errorf("%s is not a valid page number", strings.TrimPrefix(pageRangeParts[pageRangePartI], "r"))
 				}
 
 				if pageCount-parsedPageNumber < 1 || pageCount-parsedPageNumber > pageCount {
+					if ignoreInvalidPages {
+						pageNumbers = append(pageNumbers, 1)
+						continue
+					}
 					return nil, nil, fmt.Errorf("%d is not a valid page number, the document has %d page(s)", pageCount-parsedPageNumber, pageCount)
 				}
 
 				pageNumbers = append(pageNumbers, pageCount-parsedPageNumber)
 			} else {
 				parsedPageNumber, err := strconv.Atoi(pageRangeParts[pageRangePartI])
 				if err != nil {
+					if ignoreInvalidPages {
+						continue
+					}
 					return nil, nil, fmt.Errorf("%s is not a valid page number", pageRangeParts[pageRangePartI])
 				}
 
 				if parsedPageNumber < 1 || parsedPageNumber > pageCount {
+					if ignoreInvalidPages {
```

Comment:
Why add the last page here?

Reply:
If you use page range `1-10` but the file has only 5 pages, this corrects the range to `1-5`.